core/app/models/spree/order.rb

```diff
@@ -113,17 +121,22 @@ def states
 
     # Payments
     has_many :payments, dependent: :destroy, inverse_of: :order
-    has_many :valid_store_credit_payments, -> { store_credits.valid }, inverse_of: :order, class_name: 'Spree::Payment', foreign_key: :order_id
+    has_many :valid_store_credit_payments,
+      -> { store_credits.valid },
+      inverse_of: :order,
+      class_name: 'Spree::Payment',
+      foreign_key: :order_id,
+      dependent: :destroy
```

Comment:
If we do that, can we then also remove all store credit records when we delete a user (see previous comment on `UserMethods`)?